models/base.py

```diff
@@ -978,6 +981,152 @@ def calculate_coordinates(self):
         self.ecllat = coords.barycentrictrueecliptic.lat.deg
         self.ecllon = coords.barycentrictrueecliptic.lon.deg
 
+    @hybrid_method
+    def within( self, fourcorn ):
+        """An SQLAlchemy filter to find all things within a FourCorners object
+
+        Parameters
+        ----------
+          fourcorn: FourCorners
+            A FourCorners object
+
+        Returns
+        -------
+          An expression usable in a sqlalchemy filter
+
+        """
+
+        return func.q3c_poly_query( self.ra, self.dec,
+                                    sqlarray( [ fourcorn.ra00, fourcorn.dec00, fourcorn.ra01, fourcorn.dec01,
+                                                fourcorn.ra11, fourcorn.dec11, fourcorn.ra10, fourcorn.dec10 ] ) )
+
+class FourCorners:
+    """A mixin for tables that have four RA/Dec corners"""
+
+    ra00 = sa.Column( sa.REAL, nullable=False, index=True, doc="RA of the low-RA, low-Dec corner (degrees)" )
+    ra01 = sa.Column( sa.REAL, nullable=False, index=True, doc="RA of the low-RA, high-Dec corner (degrees)" )
+    ra10 = sa.Column( sa.REAL, nullable=False, index=True, doc="RA of the high-RA, low-Dec corner (degrees)" )
+    ra11 = sa.Column( sa.REAL, nullable=False, index=True, doc="RA of the high-RA, high-Dec corner (degrees)" )
+    dec00 = sa.Column( sa.REAL, nullable=False, index=True, doc="Dec of the low-RA, low-Dec corner (degrees)" )
+    dec01 = sa.Column( sa.REAL, nullable=False, index=True, doc="Dec of the low-RA, high-Dec corner (degrees)" )
+    dec10 = sa.Column( sa.REAL, nullable=False, index=True, doc="Dec of the high-RA, low-Dec corner (degrees)" )
+    dec11 = sa.Column( sa.REAL, nullable=False, index=True, doc="Dec of the high-RA, high-Dec corner (degrees)" )
```

Comment:
*We* are going to be typing it a lot... or, at least, I already did.  I'd just assume leave it as it is.  Given that there are four of them, it'll probably be pretty obvious to anybody looking at it, and the documentation does say what it is.

Reply:
It isn't too late to do a search-and-replace :) 

I think adding "corner" into those names is going to make things a lot clearer. It will be harder to do later on.